### ISSUE TITLE
Support custom generic classes

### ISSUE THREAD

**Issue by kirbycool** [AaronC81/sord#121]
Hi. If you think this is a worthwhile and feasible feature to add, I'm happy to spend some time implementing it!

I have some custom classes with generic type parameters that I'd like to be able to parse with sord. Imagine something like this.
```rb
class Wrapper
  Value = type_member
  # ...
end
```

I'd like to be able to add yardoc like this:
```rb
# @param wrapper [Wrapper<String>]
def string_unwrapper(wrapper); end
```
and get this rbi
```rb
sig { params(wrapper: Wrapper[String]) }
def string_unwrapper(wrapper); end
```

## Current behavior
Currently, sord has a hard-coded list of supported generics and errors with "unsupported generic type" if the type is not in that list.

I think it might actually possible to work around this by extending `Parlour::Types::*` with handlers for my custom types, although I haven't tried that yet. Is that the intended way to handle custom generic types?

## Implementation thoughts
We can probably implement this with the existing `Parlour::Types::Raw` by appending to appropriate sorbet type parameter syntax to the base type (e.g. `Parlour::Types::Raw.new("#{type}[#{parameters.join(', ')}]`). Alternatively, we could extend the parlour types to have a semantic understanding of type parameters, probably passing a `type_params:` option.

**Comment by AaronC81** [comment]
This sounds like a really good idea!

I like your suggestion about extending the Parlour types for this. I think the best way to do this would be to add a `Parlour::Types::Generic` class, which has a "base" type (the one you're applying the type parameters to), and then a set of type parameters. For example, your `Wrapper[String]` example would have a structure something like this:

```ruby
Generic(
  base: Raw("Wrapper"),
  type_params: [
    Raw("String")
  ]
)
```

If you make a PR to add something like that to Parlour's `rbs` branch, then I'll get it merged :+1: 

With that, you could fall back to instantiating a `Parlour::Types::Generic` if a generic type doesn't match any of Sord's built-in ones.

Thanks in advance for taking a look at this :)

**Comment by kirbycool** [comment]
@AaronC81 Thanks for the reviews! When you get a chance, can you release a new beta version of sord? Then we can close this issue.

**Comment by AaronC81** [comment]
@kirbycool I've just released 3.0.0.beta.2 which includes your changes!
